pkg/config/validation.go

```diff
@@ -0,0 +1,347 @@
+			if err != nil {
+				return fmt.Errorf("invalid exports[%d].patches[%d]: %w", idx, patchIdx, err)
+			}
+		}
+
+		for patchIdx, patch := range exp.ReversePatches {
+			err := validatePatch(patch)
+			if err != nil {
+				return fmt.Errorf("invalid exports[%d].reversPatches[%d]: %w", idx, patchIdx, err)
+			}
+		}
+	}
+
+	err = validateImportDuplicates(config.Imports)
+	if err != nil {
+		return err
+	}
+
+	for idx, imp := range config.Imports {
+		if imp == nil {
+			return fmt.Errorf("imports[%d] is required", idx)
+		}
+
+		if imp.Kind == "" {
+			return fmt.Errorf("imports[%d].kind is required", idx)
+		}
+
+		if imp.APIVersion == "" {
+			return fmt.Errorf("imports[%d].APIVersion is required", idx)
+		}
+
+		for patchIdx, patch := range imp.Patches {
+			err := validatePatch(patch)
+			if err != nil {
+				return fmt.Errorf("invalid imports[%d].patches[%d]: %w", idx, patchIdx, err)
+			}
+		}
+
+		for patchIdx, patch := range imp.ReversePatches {
+			err := validatePatch(patch)
+			if err != nil {
+				return fmt.Errorf("invalid imports[%d].reversPatches[%d]: %w", idx, patchIdx, err)
+			}
+		}
+	}
+
+	if config.Hooks != nil {
+		// HostToVirtual validation
+		for idx, hook := range config.Hooks.HostToVirtual {
+			for idy, verb := range hook.Verbs {
+				if err := validateVerb(verb); err != nil {
+					return fmt.Errorf("invalid hooks.hostToVirtual[%d].verbs[%d]: %w", idx, idy, err)
+				}
+			}
+
+			for idy, patch := range hook.Patches {
+				if err := validatePatch(patch); err != nil {
+					return fmt.Errorf("invalid hooks.hostToVirtual[%d].patches[%d]: %w", idx, idy, err)
+				}
+			}
+		}
+
+		// VirtualToHost validation
+		for idx, hook := range config.Hooks.VirtualToHost {
+			for idy, verb := range hook.Verbs {
+				if err := validateVerb(verb); err != nil {
+					return fmt.Errorf("invalid hooks.virtualToHost[%d].verbs[%d]: %w", idx, idy, err)
+				}
+			}
+
+			for idy, patch := range hook.Patches {
+				if err := validatePatch(patch); err != nil {
+					return fmt.Errorf("invalid hooks.virtualToHost[%d].patches[%d]: %w", idx, idy, err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func validatePatch(patch *config.Patch) error {
+	switch patch.Operation {
+	case config.PatchTypeRemove, config.PatchTypeReplace, config.PatchTypeAdd:
+		if patch.FromPath != "" {
+			return fmt.Errorf("fromPath is not supported for this operation")
+		}
+
+		return nil
+	case config.PatchTypeRewriteName, config.PatchTypeRewriteLabelSelector, config.PatchTypeRewriteLabelKey, config.PatchTypeRewriteLabelExpressionsSelector:
+		return nil
+	case config.PatchTypeCopyFromObject:
+		if patch.FromPath == "" {
+			return fmt.Errorf("fromPath is required for this operation")
+		}
+
+		return nil
+	default:
+		return fmt.Errorf("unsupported patch type %s", patch.Operation)
+	}
+}
+
+func validateVerb(verb string) error {
+	if !lo.Contains(verbs, verb) {
+		return fmt.Errorf("invalid verb \"%s\"; expected on of %q", verb, verbs)
+	}
```

Comment:
can't we use `slices.Contain` here?

Reply:
Does that provide any benefit?

Reply:
it's the standard library, instead of using third party one. What's the benefit of using `lo` is the real question

Reply:
Ah makes sense, will change!